src/hash.cr

```diff
@@ -782,7 +1376,7 @@ class Hash(K, V)
 
   # Returns the first key in the hash.
   def first_key
-    @first.not_nil!.key
+    first_key?.not_nil!
```

Comment:
```suggestion
    first_key?.as K
```

Reply:
That's not good because if `K` is nilable then this will succeed.

In any case thanks to your comments I realized the implementation of `first_key`, `last_key`, `first_value` and `last_value` were wrong. I fixed them.

Reply:
Why? The type of `last_key?` is `K | Nil`, so `.not_nil!` removes the `Nil` type. That's much clearer than `.as(K)`.

Reply:
@asterite you're right, we don't want to raise if the hash keys exists and are nil :+1:

Reply:
Can not nil be used as a key, anyhow?